sys/include/net/nanocoap.h

```diff
@@ -402,6 +406,14 @@ typedef struct {
     uint8_t *opt;                   /**< Pointer to the placed option       */
 } coap_block_slicer_t;
 
+#ifdef MODULE_NANOCOAP_SERVER_XFA
+/**
+ * @brief   CoAP XFA resource entry
+ *
+ * @param name  internal name of the resource entry, must be unique
+ */
+#define ADD_COAP_RESOURCE(name) XFA_CONST(coap_resources_xfa, 0) coap_resource_t name =
```

Comment:
```suggestion
#define ADD_COAP_RESOURCE(name) XFA_CONST(coap_resources_xfa, 0) coap_resource_t CONCAT(coap_resource_, name) =
```

Maybe prefix to avoid name clashes?